webhooks/change/terraform-aws-github-files-webhook/src/index.ts

```diff
@@ -63,43 +65,67 @@ const octokit = new Octokit({
   auth: GITHUB_TOKEN,
 })
 
-async function getFile({ owner, repo, path }) {
-  return await octokit.repos
-    .getContent({
-      owner,
-      repo,
-      path,
-    })
-    .then((r) => r.data)
+async function getFile({
+  owner,
+  repo,
+  path,
+}: {
+  owner: string
+  repo: string
+  path: string
+}) {
+  const repoContents = await octokit.repos.getContent({ owner, repo, path })
+  if ('content' in repoContents.data) {
+    if (Array.isArray(repoContents.data.content)) {
```

Comment:
I think this should type-guard `repoContents.data` rather than `repoContents.data.content`; judging by [the GitHub example responses](https://docs.github.com/en/rest/reference/repos#get-repository-content), the `content` field is only present when `path` points to a file, and it's always the file contents as an (encoded) string.

Instead:

```ts
const repoContents = await octokit.repos.getContent({ owner, repo, path })
if (Array.isArray(repoContents.data) {
  // throw error: it's a directory.
}
switch (repoContents.data.type) {
  case 'file':
    return repoContents.data
  default:
    // Throw some error, including repoContents.data.type
}
```

Reply:
could this just check `if (repoContents.data?.type !== 'file') {` and throw?

Reply:
After excluding the array type from the `repoContents.data` union type, yes!

Reply:
@fouad per our huddle that's why it didn't work to just do if (repoContents.data.content?) because it's a complex union type